include/dpp/isa_detection.hpp

```diff
@@ -0,0 +1,256 @@
+/************************************************************************************
+ *
+ * D++, A Lightweight C++ library for Discord
+ *
+ * Copyright 2021 Craig Edwards and D++ contributors
+ * (https://github.com/brainboxdotcc/DPP/graphs/contributors)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ************************************************************************************/
+#pragma once
+
+#if defined _MSC_VER || defined __GNU__ || defined __clang__
+	#include <immintrin.h>
+#endif
+
+#ifdef HAVE_VOICE
+	#include <opus/opus.h>
```

Comment:
we shouldnt include this in the public header, or it makes the bot diretly depend on opus headers. This can cause huge issues. Can this be put inside the implementation without causing problems? opus_int16 etc are basically exactly like stdint variants of the same thing.